src/client/Microsoft.Identity.Client/ManagedIdentity/ManagedIdentitySource.cs

```diff
@@ -29,14 +30,31 @@ protected ManagedIdentitySource(RequestContext requestContext)
 
         public virtual async Task<ManagedIdentityResponse> AuthenticateAsync(AppTokenProviderParameters parameters, CancellationToken cancellationToken)
         {
-            ManagedIdentityRequest request = CreateRequest(parameters.Scopes.ToArray());
+            // Convert the scopes to a resource string.
+            string resource = ScopeHelper.ScopesToResource(parameters.Scopes.ToArray());
 
-            HttpResponse response =
+            if (cancellationToken.IsCancellationRequested)
```

Comment:
Shouldn't this also be higher than the ScopesToResource(). Should probably be the first thing checked for.